src/sql/workbench/services/query/common/queryRunner.ts

```diff
@@ -276,6 +280,17 @@ export default class QueryRunner extends Disposable {
 		this._onMessage.fire([message]);
 	}
 
+	/**
+	 * Handle a QueryComplete from the service layer
+	 */
+	public handlePid(pid: string): void {
+		let pidEvent: QueryEvent = {
+			type: this.uri,
```

Comment:
The type is the URI? That doesn't seem right. You should probably just make your own interface type to fire for the event instead of re-using this one.